letsencrypt/client/tests/crypto_util_test.py

```diff
@@ -133,5 +134,71 @@ def test_san(self):
         self._call('cert-san.pem')
 
 
+class GetSansFromCsrTest(unittest.TestCase):
+    """Tests for letsencrypt.client.crypto_util.get_sans_from_csr."""
+    def test_extract_one_san(self):
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        csr = pkg_resources.resource_string(
+            __name__, os.path.join('testdata', 'csr.pem'))
+        result = get_sans_from_csr(csr)
+        self.assertEqual(result, ['example.com'])
+
+    def test_extract_two_sans(self):
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        csr = pkg_resources.resource_string(
+            __name__, os.path.join('testdata', 'csr-san.pem'))
+        result = get_sans_from_csr(csr)
+        self.assertEqual(result, ['example.com', 'www.example.com'])
+
+    def test_extract_six_sans(self):
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        csr = pkg_resources.resource_string(
+            __name__, os.path.join('testdata', 'csr-6sans.pem'))
+        result = get_sans_from_csr(csr)
+        self.assertEqual(
+            result, ["example.com", "example.org", "example.net",
+                     "example.info", "subdomain.example.com",
+                     "other.subdomain.example.com"])
+
+    def test_parse_non_csr(self):
+        from M2Crypto.X509 import X509Error
+        from letsencrypt.client.crypto_util import get_sans_from_csr
+        self.assertRaises(X509Error, get_sans_from_csr, "hello there")
+
+    def test_parse_no_sans(self):
```

Comment:
It should be documented that `get_sans_from_csr` requires a SAN CSR and fails otherwise (in `get_sans_from_csr.__doc__`)